javascript/app.js

```diff
@@ -0,0 +1,254 @@
+}
+console.log(sumDigits(42));
+// => 6;
+
+// 🔴 **Commit your work.** 
+// The commit message should read: 
+// "Commit 3 - Digit Sum".
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+// 4. Pythagoras
+// Write a function calculateSide that takes two arguments: sideA and sideB, and returns the solution for sideC using the Pythagorean theorem.
+const calculateSide = (sideA, sideB) => {
+	return Math.sqrt(Math.pow(sideA, 2) + Math.pow(sideB, 2));
+}
+
+console.log(calculateSide(8, 6));
+// => 10
+
+// 🔴 **Commit your work.** 
+// The commit message should read: 
+// "Commit 4 - Pythagoras".
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+// 5. Sum Array
+// Write a function sumArray that takes an array as an argument. The array should contain numbers. The function should return the sum of the numbers in the array.
+const sumArray = (array) => {
+	// From what I understand .reduce() makes us able to iterate through an array
+	// According to MDN .reduce() "executes a reducer function [...] on each member of the array resulting in a single output value."
+	return array.reduce((a, b) => a + b, 0)
+}
+
+// Expected result:
+console.log(sumArray([1, 2, 3, 4, 5, 6]));
+// => 21
+
+// 🔴 **Commit your work.** 
+// The commit message should read: 
+// "Commit 5 - Sum Array".
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+// 6. Prime Numbers
+// A Prime number is a number that is not evenly divisible by another number except 1 and itself. If you want to read more deeply about it, go here. To test whether a number is Prime, you only need to test as far as the square root of that number. This is advisable for optimization and testing large numbers.
+
+// Step One
+// Write a function called checkPrime that will test whether a number is Prime. The function will return true (Boolean) if Prime, false if not. Hint: Check every number up to the square root. To do this, try a for loop.
+// const checkPrime = (num) => {
+// 	// Start with 2 because 1 is its own special special case
+// 	if (num === 1) {
+// 		return `1 is a special case!`;
+// 	}
+// 	for (let i = 2; i < num; i++) {
+// 		if (num % i === 0) {
+// 			return false;
+// 		}
+// 		return num !== 1;
+// 	}
+// }
+// console.log(checkPrime(13));
+
+// Step Two
+// Write another function called printPrimes that will print (console log) all the Primes up to an arbitrary limit. For example, if you invoke your function with printPrimes(97), it will print all the Prime numbers up to and including 97. This function can call on the previous checkPrime function.
+
+const checkPrime = (num) => {
+	// Start with 2 because 1 is its own special special case
+	if (num === 1) {
+		return `1 is a special case!`;
+	}
+	for (let i = 2; i < num; i++) {
+		if (num % i === 0) {
+			return false;
+		}
+	}
+	return num !== 1;
+
+}
```

Comment:
a function called `check`something typically should only return booleans, not a string "1 is a special case"

but you have the right idea by accounting for edge cases, just return false instead of the string.